src/xunit.performance.api/XunitPerformanceHarness.cs

```diff
@@ -1,88 +1,69 @@
+using MarkdownLog;
+using Microsoft.Xunit.Performance.Api.Table;
 using System;
 using System.Collections.Generic;
+using System.Globalization;
 using System.IO;
 using System.Linq;
-using System.Reflection;
-using Microsoft.Xunit.Performance;
-using Microsoft.Xunit.Performance.Api.Table;
+using System.Xml.Linq;
+using static Microsoft.Xunit.Performance.Api.XunitPerformanceLogger;
 
 namespace Microsoft.Xunit.Performance.Api
 {
     public sealed class XunitPerformanceHarness : IDisposable
     {
-        private string[] _args;
-
         public XunitPerformanceHarness(string[] args)
         {
             _args = args;
+            _disposed = false;
 
             // Set the run id.
-            Configuration.RunId = DateTimeOffset.UtcNow.ToString("yyyy-MM-dd-HH-mm-ss");
+            Configuration.RunId = $"{DateTimeOffset.UtcNow.ToString("yyyyMMddHHmmss")}";
 
             // Set the file log path.
             // TODO: Conditionally set this based on whether we want a csv file written.
             Configuration.FileLogPath = Configuration.RunId + ".csv";
         }
 
-        void IDisposable.Dispose()
+        public BenchmarkConfiguration Configuration
         {
-            // Close the log when all test cases have completed execution.
-            // TODO: This is a hack because we haven't found a way to close the file from within xunit.
-            BenchmarkEventSource.Log.Close();
-
-            // Process the results now that we know we're done executing tests.
-            ProcessResults();
+            get { return BenchmarkConfiguration.Instance; }
         }
 
-        public BenchmarkConfiguration Configuration
+        public int RunBenchmarks(string assemblyPath)
         {
-            get { return BenchmarkConfiguration.Instance; }
+            Validate(assemblyPath);
+            int errCode = 0;
+            ETWProfiler.Profile(assemblyPath, Configuration.RunId, () => { errCode = XunitRunner.Run(assemblyPath); });
+            return errCode;
         }
 
-        public void RunBenchmarks(string assemblyPath)
+        private static void Validate(string assemblyPath)
         {
             if (string.IsNullOrEmpty(assemblyPath))
-            {
                 throw new ArgumentNullException(nameof(assemblyPath));
-            }
-
-            if(!File.Exists(assemblyPath))
-            {
+            if (!File.Exists(assemblyPath))
                 throw new FileNotFoundException(assemblyPath);
-            }
-
-            // Invoke xunit to run benchmarks in the specified assembly.
-            XunitRunner runner = new XunitRunner();
-            runner.Run(assemblyPath);
         }
 
         private void ProcessResults()
         {
-            CSVMetricReader reader = new CSVMetricReader(Configuration.FileLogPath);
-            WriteStatisticsToFile(reader);
+            var reader = new CSVMetricReader(Configuration.FileLogPath);
```

Comment:
The whole file output produced by the API needs some cleaning.
Currently, the ETW class does a checking on the OS, and only captures ETW on Windows.

---
In reply to: [100352115](https://github.com/Microsoft/xunit-performance/pull/183#discussion_r100352115) [](ancestors = 100352115)